File: src/app/Models/product.ts
export class Product {

    ProductId: number;
    Name: string;
    Price: number;
    Country: string;
    ProductTypeId: number;

    constructor (id: number, _name: string, _price: number, _country: string, _ProductTypeId: number){
        this.ProductId = id;
        this.Name = _name;
        this.Price = _price;
        this.Country = _country;
    }
  }

File: ProductsAPI/ProductsData/Entities/Product.cs
using Microsoft.EntityFrameworkCore.Metadata.Internal;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProductsData.Entities
{
  public class Product
  {
    [Key]
    public int productId { get; set; }
    public string? name { get; set; }  
    public int price { get; set; }
    public string? country { get; set; }
    public ProductType? productType { get; set; }
  }
}
